broadinstitute/single_cell_portal_core
# Retiring obsolete 'faceted_search' feature flag (SCP-3621)

Similar to #1145, this update removes the obsolete `faceted_search` feature flag.  In addition to retiring & unsetting feature flag values, this also removes the "advanced search" opt-in/out behavior.

This also includes a small enhancement to update the advanced search help modal to dynamically update the statistics regarding the number of public, conventional SCP studies.  This functionality has been added to `SummaryStatsUtils` in case we wish to include this information elsewhere.

MANUAL TESTING
* Run `ruby_local_setup.rb && source config/secrets/.source_env.bash && rails db:migrate`
* Boot portal as normal
* On the home page, note that the `?` link next to `More Facets` no longer contains opt in/out buttons when opened
* Load the Admin Config page, and edit the feature flags for your account
* Note that the form renders w/o error, and that the `faceted_search` entry is gone
* Optional: in the Rails console, query for any user accounts that still have a value set for `feature_flags.faceted_search` and confirm the return is 0:
```
User.where(:'feature_flags.faceted_search'.ne => nil).count
=> 0
```

This PR satisfies SCP-3621.
